src/framework/scene-registry.js

```diff
@@ -253,89 +258,96 @@ class SceneRegistry {
-                self._app.systems.fire('postInitialize', entity);
-                self._app.systems.fire('postPostInitialize', entity);
+                app.systems.fire('initialize', entity);
+                app.systems.fire('postInitialize', entity);
+                app.systems.fire('postPostInitialize', entity);
 
-                if (callback) callback(err, entity);
+                if (callback) callback(null, entity);
             };
 
             // load priority and referenced scripts before opening scene
-            self._app._preloadScripts(data, _loaded);
+            app._preloadScripts(sceneItem.data, _loaded);
         });
     }
 
+    /**
+     * Load a scene file, create and initialize the Entity hierarchy and add the hierarchy to the
+     * application root Entity.
+     *
+     * @param {SceneRegistryItem | string} sceneItem - The scene item (which can be found with
+     * {@link SceneRegistry#find}, URL of the scene file (e.g."scene_id.json") or name of the scene.
+     * @param {LoadHierarchyCallback} callback - The function to call after loading,
+     * passed (err, entity) where err is null if no errors occurred.
+     * @example
+     * var sceneItem = app.scenes.find("Scene Name");
+     * app.scenes.loadSceneHierarchy(sceneItem, function (err, entity) {
+     *     if (!err) {
+     *         var e = app.root.find("My New Entity");
+     *     } else {
+     *         // error
+     *     }
+     * });
+     */
+    loadSceneHierarchy(sceneItem, callback) {
+        this._loadSceneHierarchy(sceneItem, null, callback);
+    }
+
     /**
      * Load a scene file and apply the scene settings to the current scene.
      *
      * @param {SceneRegistryItem | string} sceneItem - The scene item (which can be found with
-     * {@link SceneRegistry#find} or URL of the scene file. Usually this will be "scene_id.json".
+     * {@link SceneRegistry#find}, URL of the scene file (e.g."scene_id.json") or name of the scene.
      * @param {LoadSettingsCallback} callback - The function called after the settings
      * are applied. Passed (err) where err is null if no error occurred.
      * @example
      * var sceneItem = app.scenes.find("Scene Name");
-     * app.scenes.loadSceneHierarchy(sceneItem, function (err, entity) {
+     * app.scenes.loadSceneSettings(sceneItem, function (err) {
      *     if (!err) {
-     *         var e = app.root.find("My New Entity");
+     *         // success
      *     } else {
      *         // error
      *     }
      * });
      */
     loadSceneSettings(sceneItem, callback) {
-        const self = this;
+        const app = this._app;
 
-        this._loadSceneData(sceneItem, false, function (err, sceneItem) {
+        this._loadSceneData(sceneItem, false, (err, sceneItem) => {
             if (!err) {
-                self._app.applySceneSettings(sceneItem.data.settings);
+                app.applySceneSettings(sceneItem.data.settings);
```

Comment:
Since you are using arrow functions now, you can just use `this` and move that closure variable into the arrow function itself. And since `app` is only used once, it doesn't really need to be declared at all